### PR TITLE
zabbix: add variants for SSL support

### PR DESCRIPTION
**Pull request by ttocsr**
opkg does not offer ssl varients of zabbix-agentd. resolve this by adding ssl varients.

Compile tested: aarch64 cortex-a53
Run tested: aarch64 cortex-a53

Signed-off-by: Scott Roberts <ttocsr@gmail.com>

Supersedes: https://github.com/openwrt/packages/pull/17846

### CODE REVIEW

**Comment by feckert** [comment]
I have tested your PR in my own build.
I get now the following error message:
```
Package zabbix-sender is missing dependencies for the following libraries:
libcrypto.so.1.1
libssl.so.1.1
```

So I think we also need to add a build variant for zabbix-sender.
```
diff --git a/admin/zabbix/Makefile b/admin/zabbix/Makefile
index 19c2a898e23317b2ee8bdb6631e88202879c91d1..7a0b615eaa21cca012b8b850574517aaf152f5a2 100644
--- a/admin/zabbix/Makefile
+++ b/admin/zabbix/Makefile
@@ -101,6 +101,25 @@ endef
 define Package/zabbix-sender
   $(call Package/zabbix/Default)
   TITLE+= sender
+  PROVIDES:=zabbix-sender
+  VARIANT:=nossl
+  DEFAULT_VARIANT:=1
+endef
+
+define Package/zabbix-sender-openssl
+  $(call Package/zabbix/Default)
+  TITLE+= sender (with OpenSSL)
+  DEPENDS+= +libopenssl
+  PROVIDES:=zabbix-sender
+  VARIANT:=openssl
+endef
+
+define Package/zabbix-sender-gnutls
+  $(call Package/zabbix/Default)
+  TITLE+= sender (with GnuTLS)
+  DEPENDS+= +libgnutls
+  PROVIDES:=zabbix-sender
+  VARIANT:=gnutls
 endef

 define Package/zabbix-get
@@ -300,6 +319,8 @@ endef
 define Package/zabbix-sender/install
        $(call Package/zabbix/install/bin,$(1),sender)
 endef
+Package/zabbix-sender-openssl/install = $(Package/zabbix-sender/install)
+Package/zabbix-sender-gnutls/install = $(Package/zabbix-sender/install)

 define Package/zabbix-get/install
        $(call Package/zabbix/install/bin,$(1),get)
@@ -327,6 +348,8 @@ $(eval $(call BuildPackage,zabbix-extra-mac80211))
 $(eval $(call BuildPackage,zabbix-extra-network))
 $(eval $(call BuildPackage,zabbix-extra-wifi))
 $(eval $(call BuildPackage,zabbix-sender))
+$(eval $(call BuildPackage,zabbix-sender-openssl))
+$(eval $(call BuildPackage,zabbix-sender-gnutls))
 $(eval $(call BuildPackage,zabbix-server))
 $(eval $(call BuildPackage,zabbix-server-frontend))
 $(eval $(call BuildPackage,zabbix-proxy))
--
```

**Comment by ttocsr** [comment]
zabbix-sender has never had ssl support in the make file.
I'm not sure why you're getting that error that ssl is a dependency. 
Are you just looking to also add ssl support for zabbix-sender like we do for zabbix-agentd?

**Comment by feckert** [comment]
You're right about that.
I had another commit in my patch set.
From my point of view, I would say that this works and could be merged.
If there are no objections from maintainer of this package @champtar , I would merge this in the next few days.

We can then continue with the following variants:
* Add ssl variant also for zabbix-sender, zabbix-proxy and zabbix-server
* Add variants for zabbix-server and zabbix-proxy with the SQL backen (postgres and mysql) are also interesting
* Add a variant for zabbix-proxy with sqlite3 backend as I know this does not work for zabbix-server

The SQL variants could be problematic since there are already the individual ssl variants!

**Comment by ttocsr** [comment]
Does it make sense for me to just add all the ssl variants here in this PR?
Then you could add the PR for SQL work?

**Comment by champtar** [comment]
@feckert ack to merge this when you feel it's ready

**Comment by ttocsr** [comment]
@feckert let me know if I missed something.

File: admin/zabbix/Makefile
Comment (feckert) [inline admin/zabbix/Makefile:86]: That's a small thing and a nitpick from my side but, this is not part of the commit and is not necessary, so either don't do it or outsource it to a separate commit.

File: admin/zabbix/Makefile
Comment (feckert) [inline admin/zabbix/Makefile:92]: See my comment above

File: admin/zabbix/Makefile
Comment (feckert) [inline admin/zabbix/Makefile:98]: See my comment above

File: admin/zabbix/Makefile
Comment (feckert) [inline admin/zabbix/Makefile:153]: See my comment above

File: admin/zabbix/Makefile
Comment (feckert) [inline admin/zabbix/Makefile:184]: See my comment above

File: admin/zabbix/Makefile
Comment (feckert) [inline admin/zabbix/Makefile:215]: Move this is a separate commit. See my comment above.

File: admin/zabbix/Makefile
Comment (feckert) [inline admin/zabbix/Makefile:196]: The server frontend is a PHP application. And as I understand it, there is no difference between openssl and gnutls. If there is a variant needed, then the only difference is whether we use MySQL or PostgreSQL (but this is the next setup on update zabbix)

**Review by feckert — CHANGES_REQUESTED** [review]
Looks pretty good so far. Please have a look at my comments

**Review by feckert — APPROVED** [review]
Very nice. Works as expected :+1:

**Pull request merged**
